### PR TITLE
OVDB-99: Fix an integer Tuple/Mat compile bug on Windows 

### PR DESCRIPTION
**Pull request by danrbailey**

### CODE REVIEW

File: openvdb/CHANGES [hunk 1]
```diff
@@ -15,6 +15,10 @@ Version 6.1.1 - In development
     - Fixed a bug in tools::computeScalarPotential() that could produce
       a corrupt result due to invalid memory access.
       [Reported by Edwin Braun]
+    - Fixed a bug in isInfinite() and isNan() methods of math::Tuple and
```
Comment (pcucka) [inline openvdb/CHANGES]: This is a workaround for a bug in VS, not a bug in OpenVDB.  C++11 requires `std::isfinite` and `std::isnan` to be overloaded for integral types.

Comment (danrbailey) [reply openvdb/CHANGES]: To clarify - you're happy with the change itself but you'd like it to be recorded as a VS workaround in the release notes as opposed to a VDB bug?

Comment (pcucka) [reply openvdb/CHANGES]: I would prefer that we deprecate functions like `math::isFinite`, `math::Sqrt`, etc. for which there are standard library equivalents, but it sounds like that's not going to be possible any time soon.

Comment (danrbailey) [reply openvdb/CHANGES]: It's still an issue in Visual Studio 2019, so no, I don't think this is going away any time soon.

I've improved the change notes.

**Comment by danrbailey** [comment]
I just merged master in and fixed the conflicts in the release notes. @pcucka - can we merge this?

File: openvdb/CHANGES [hunk 2]
```diff
@@ -12,6 +12,8 @@ Version 6.1.1 - In development
       now modify voxel values in place for improved performance.
     - Add CMake support for linking against Jemalloc and TBB malloc and enable
       Jemalloc by default for Linux and non-Maya, otherwise use TBB malloc.
+    - Made a few improvements related to the use of std::inf and std::nan with
```
Comment (pcucka) [inline openvdb/CHANGES]: How about "Added math::isInfinite() and math::isNaN() to resolve Visual Studio compatibility issues with integer types"?

File: openvdb/math/Math.h [hunk 3]
```diff
@@ -351,6 +351,26 @@ inline bool
...
+
+/// Return @c true if @a x is an infinity value (either positive infinity or negative infinity).
+template<typename Type, typename std::enable_if<std::is_arithmetic<Type>::value, int>::type = 0>
+inline bool
+isInfinite(const Type& x) { return std::isinf(static_cast<double>(x)); }
+
+
+/// Return @c true if @a x is a NaN (Not-A-Number) value.
+inline bool
+isNan(const float x) { return std::isnan(x); }
```
Comment (pcucka) [inline openvdb/math/Math.h]: Let's call this "isNaN".

File: openvdb/CHANGES (hunk 2 above)
Comment (danrbailey) [reply openvdb/CHANGES]: Sure

File: openvdb/math/Math.h (hunk 3 above)
Comment (danrbailey) [reply openvdb/math/Math.h]: I'm not keen on the inconsistency between math::isNaN() and math::Tuple::isNan(), but that's a minor thing. I'll make that change now.

Comment (pcucka) [reply openvdb/math/Math.h]: You're right, they should be consistent.  But I think I would prefer that we add `Tuple::isNaN` and `Mat::isNaN` and deprecate the `isNan` methods.

Comment (danrbailey) [reply openvdb/math/Math.h]: Do you not think we might annoy users forcing them to change their code when they update simply to capitalize one letter? The current capitalization is at least consistent with Half:

https://github.com/openexr/openexr/blob/master/IlmBase/Half/half.h#L171

Comment (danrbailey) [reply openvdb/math/Math.h]: I changed it back to isNan so this is consistent and users don't have to change their code.

**Pull request merged**